### PR TITLE
feat: add configuration java support

### PR DESCRIPTION
**Pull request by qingzhuozhen**
### Summary

- update configuration to support better in java
- add example to init optionally with configuration

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Kotlin/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

File: core/src/main/java/com/amplitude/core/Configuration.kt [hunk 1]
```diff
@@ -8,21 +8,21 @@ import com.amplitude.core.utilities.InMemoryStorageProvider
 typealias EventCallBack = (BaseEvent, status: Int, message: String) -> Unit
 
 open class Configuration @JvmOverloads constructor(
-    val apiKey: String,
-    var flushQueueSize: Int = FLUSH_QUEUE_SIZE,
-    var flushIntervalMillis: Int = FLUSH_INTERVAL_MILLIS,
-    var instanceName: String = DEFAULT_INSTANCE,
-    var optOut: Boolean = false,
-    val storageProvider: StorageProvider = InMemoryStorageProvider(),
```
Comment (yuhao900914) [inline core/src/main/java/com/amplitude/core/Configuration.kt:16]: Should this be va r or val? do we allowed to override?

Comment (qingzhuozhen) [reply core/src/main/java/com/amplitude/core/Configuration.kt:16]: Good point, I was wondering about this as well. That time I changed to var to enable it to be set, taking another thought, maybe we can use this as a val first until it is need in the future. Let me change it back to val for now.

File: android/src/main/java/com/amplitude/android/Configuration.kt [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
 import com.amplitude.core.Configuration
 import com.amplitude.core.EventCallBack
 import com.amplitude.core.LoggerProvider
 import com.amplitude.core.ServerZone
 import com.amplitude.core.StorageProvider
+import com.amplitude.core.events.Plan
 
 class Configuration @JvmOverloads constructor(
-    apiKey: String,
+    override val apiKey: String,
```
Comment (yuhao900914) [inline android/src/main/java/com/amplitude/android/Configuration.kt:14]: why we use override here for api key?

Comment (qingzhuozhen) [reply android/src/main/java/com/amplitude/android/Configuration.kt]: Good catch, that was accidentally added..

**Pull request merged**